### PR TITLE
d/jak2: finish `viewer`, `texture-finish`, `process-taskable`, `main-collide`, `prototype` and `los-control`

### PR DESCRIPTION
**Pull request by xTVaser**

### CODE REVIEW

File: goal_src/jak2/engine/debug/viewer.gc [hunk 1]
```diff
@@ -5,5 +5,262 @@
...
 ;; dgos: ENGINE, GAME
 
+(declare-type viewer process-drawable)
+(define-extern *viewer* viewer)
+
 ;; DECOMP BEGINS
 
+(define *viewer-sg* (new 'static 'skeleton-group
+                      :bounds (new 'static 'vector :w 16384.0)
+                      :lod-dist (new 'static 'array float 6 4095996000.0 0.0 0.0 0.0 0.0 0.0)
```
Comment (water111) [inline goal_src/jak2/engine/debug/viewer.gc:15]: I think the skelgroup stuff is merged - does this get converted to `defskelgroup` on the latest master?

File: goal_src/jak2/engine/gfx/texture/texture-finish.gc [hunk 2]
```diff
@@ -7,37 +7,22 @@
...
+  (when v1-2
+    (set! (-> v1-2 w) 32)
+    (set! (-> v1-2 h) 32)
+    (set! (-> v1-2 dest 0) (-> *skull-gem-texture-base* vram-block))
+    )
+  )
 
-; ;; failed to figure out what this is:
-; ((method-of-object *sky-texture-anim-array* texture-anim-array-method-9))
+((method-of-object *sky-texture-anim-array* texture-anim-array-method-9))
```
Comment (water111) [inline goal_src/jak2/engine/gfx/texture/texture-finish.gc:20]: missing args, I'd also assume these crash?

File: test/decompiler/reference/jak2/engine/collide/main-collide_REF.gc [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    (.add.w.vf vf11 vf0 vf9 :mask #b1)
+    (nop!)
+    (.mov a3-0 vf11)
+    (nop!)
+    (let ((a2-0 (new-stack-vector0)))
+      (.svf (&-> a2-0 quad) vf9)
+      (add-debug-sphere #t (bucket-id debug2) a2-0 a3-0 (new 'static 'rgba :r #x80 :g #x80 :b #x80 :a #x80))
+      )
+    )
+  )
```
Comment (water111) [inline test/decompiler/reference/jak2/engine/collide/main-collide_REF.gc:197]: missing from `goal_src`?

File: test/decompiler/reference/jak2/engine/gfx/background/prototype_REF.gc [hunk 4]
```diff
@@ -0,0 +1,114 @@
+;;-*-Lisp-*-
+(in-package goal)
+
+;; definition for method 9 of type prototype-inline-array-shrub
```
Comment (water111) [inline test/decompiler/reference/jak2/engine/gfx/background/prototype_REF.gc:4]: missing from `goal_src`?

File: goal_src/jak2/engine/debug/viewer.gc (hunk 1 above)
Comment (Hat-Kid) [reply goal_src/jak2/engine/debug/viewer.gc:15]: I checked and it doesn't - this seems to be a case where the static data is completely empty besides the `bounds` and `lod-dist` and the only thing that is defined in the `top-level` is the `skeleton-group` itself, there is no `let`, so the decompiler doesn't try to rewrite it.
The IR for Jak 1 looks exactly the same.

**Pull request merged**